packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,577 @@
+        width: 24.0,
+        height: 24.0,
+        child: container
+      )
+    );
+  }
+
+  Widget _buildVerticalControls() {
+    return new Container(
+      margin: const EdgeInsets.only(top: 16.0),
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints.tightFor(height: 48.0),
+        child: new Row(
+          children: <Widget>[
+            new FlatButton(
+              onPressed: config.onStepContinue,
+              color: Theme.of(context).primaryColor,
+              textColor: Colors.white,
+              textTheme: ButtonTextTheme.normal,
+              child: new Text('CONTINUE')
+            ),
+            new Container(
+              margin: const EdgeInsets.only(left: 8.0),
+              child: new FlatButton(
+                onPressed: config.onStepCancel,
+                textColor: Colors.black54,
+                textTheme: ButtonTextTheme.normal,
+                child: new Text('CANCEL')
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+
+  TextStyle _titleStyle(int index) {
+    switch (config.steps[index].state) {
+      case StepState.disabled:
+        return _kDisabledTitleStyle;
+      case StepState.active:
+      case StepState.inactive:
+        return _kActiveTitleStyle;
+      case StepState.error:
+        return _kErrorTitleStyle;
+      default:
+        // Unreachable.
+        return null;
+    }
+  }
+
+  TextStyle _subtitleStyle(int index) {
+    switch (config.steps[index].state) {
+      case StepState.disabled:
+        return _kDisabledSubtitleStyle;
+      case StepState.active:
+      case StepState.inactive:
+        return _kActiveSubtitleStyle;
+      case StepState.error:
+        return _kErrorSubtitleStyle;
+      default:
+        // Unreachable.
+        return null;
+    }
+  }
+
+  Widget _headerText(int index) {
+    List<Widget> children = <Widget>[];
+
+    children.add(
+      new Text(
+        config.steps[index].title,
+        style: _titleStyle(index)
+      )
+    );
+
+    if (config.steps[index].subtitle != null)
+      children.add(
+        new Container(
+          margin: const EdgeInsets.only(top: 2.0),
+          child: new Text(
+            config.steps[index].subtitle,
+            style: _subtitleStyle(index)
+          )
+        )
+      );
+
+    return new Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
+      children: children
+    );
+  }
+
+  Widget _buildVerticalHeader(BuildContext context, int index) {
+    return new Container(
+      margin: const EdgeInsets.symmetric(horizontal: 24.0),
+      child: new Row(
+        children: <Widget>[
+          new Column(
+            children: <Widget>[
+              // Line parts are always added in order for the ink splash to
+              // flood the tips of the connector lines. This solution is far
+              // simple than having to compute line positions and lengths within
```

Comment:
no need to say what it's simpler than, there's lots of more complicated solutions and we don't list them all :-)